flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/logical/FlinkAggregate.scala

```diff
@@ -58,19 +58,23 @@ class FlinkAggregate(
     )
   }
 
-  override def computeSelfCost (planner: RelOptPlanner): RelOptCost = {
-
-    val origCosts = super.computeSelfCost(planner)
-    val deltaCost = planner.getCostFactory.makeHugeCost()
-
-    // only prefer aggregations with transformed Avg
-    aggCalls.toList.foldLeft[RelOptCost](origCosts){
-      (c: RelOptCost, a: AggregateCall) =>
-        if (a.getAggregation.isInstanceOf[SqlAvgAggFunction]) {
-          c.plus(deltaCost)
-        } else {
-          c
-        }
-    }
-  }
+//
+// DO NOT ASSIGN HUGE COSTS TO PLANS WITH AVG AGGREGATIONS
+//   ONLY NECESSARY IF AggregateReduceFunctionsRule IS ENABLED.  
+//  
+//  override def computeSelfCost (planner: RelOptPlanner): RelOptCost = {
+//
+//    val origCosts = super.computeSelfCost(planner)
+//    val deltaCost = planner.getCostFactory.makeHugeCost()
+//
+//    // only prefer aggregations with transformed Avg
+//    aggCalls.toList.foldLeft[RelOptCost](origCosts){
+//      (c: RelOptCost, a: AggregateCall) =>
+//        if (a.getAggregation.isInstanceOf[SqlAvgAggFunction]) {
+//          c.plus(deltaCost)
+//        } else {
+//          c
+//        }
+//    }
+//  }
```

Comment:
Yes, we decided to disable `AggregateReduceFunctionsRule` temporarily, thus the note on top.
I could remove it completely if you think it'd be better.